pkg/controller/kubelet-config/kubelet_config_features.go

```diff
@@ -0,0 +1,207 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+
+	osev1 "github.com/openshift/api/config/v1"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	mtmpl "github.com/openshift/machine-config-operator/pkg/controller/template"
+	"github.com/openshift/machine-config-operator/pkg/version"
+)
+
+///////////////////////////////////////////////////////////////////////////////
+// FeatureGates
+///////////////////////////////////////////////////////////////////////////////
+
+const (
+	clusterFeatureInstanceName = "cluster"
+)
+
+func (ctrl *Controller) featureWorker() {
+	glog.Infof("FeatureWorker")
+	for ctrl.processNextFeatureWorkItem() {
+	}
+}
+
+func (ctrl *Controller) processNextFeatureWorkItem() bool {
+	key, quit := ctrl.featureQueue.Get()
+	if quit {
+		return false
+	}
+	defer ctrl.featureQueue.Done(key)
+
+	err := ctrl.syncFeatureHandler(key.(string))
+	ctrl.handleFeatureErr(err, key)
+	return true
+}
+
+func (ctrl *Controller) syncFeatureHandler(key string) error {
+	startTime := time.Now()
+
+	glog.V(4).Infof("Started syncing feature handler %q (%v)", key, startTime)
+	defer func() {
+		glog.V(4).Infof("Finished syncing feature handler %q (%v)", key, time.Since(startTime))
+	}()
+
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	// Fetch the Feature
+	features, err := ctrl.featLister.Get(name)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("FeatureSet %v has been deleted", key)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	featureGates, err := ctrl.generateFeatureMap(features)
+
+	// Find all MachineConfigPools
+	mcpPools, err := ctrl.mcpLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range mcpPools {
+		role := pool.Name
+
+		// Get MachineConfig
+		managedKey := getManagedFeaturesKey(pool)
+		mc, err := ctrl.client.Machineconfiguration().MachineConfigs().Get(managedKey, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		isNotFound := errors.IsNotFound(err)
+		// Generate the original KubeletConfig
+		originalKubeletIgn, err := ctrl.generateOriginalKubeletConfig(role)
+		if err != nil {
+			return err
+		}
+		dataURL, err := dataurl.DecodeString(originalKubeletIgn.Contents.Source)
+		if err != nil {
+			return err
+		}
+		originalKubeConfig, err := decodeKubeletConfig(dataURL.Data)
+		if err != nil {
+			return err
+		}
+		// Merge in Feature Gates
+		err = mergo.Merge(&originalKubeConfig.FeatureGates, featureGates, mergo.WithOverride)
+		if err != nil {
+			return err
+		}
+		// Encode the new config into YAML
+		cfgYAML, err := encodeKubeletConfig(originalKubeConfig, kubeletconfigv1beta1.SchemeGroupVersion)
+		if err != nil {
+			return err
+		}
+		if isNotFound {
```

Comment:
it was confusing to me to have this way down here.  can you move it up to line 93?

Reply:
Fixed